arviz/plots/dotplot.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+
+from ..rcparams import rcParams
+from .plot_utils import get_plotting_function
+
+
+def plot_dot(
+    values=None,
+    binwidth=None,
+    dotsize=1,
+    stackratio=1,
+    hdi_prob=None,
+    rotated=False,
+    dotcolor="C0",
+    intervalcolor="C3",
+    markersize=None,
+    markercolor="C0",
+    marker="o",
+    figsize=None,
+    linewidth=None,
+    point_estimate="auto",
+    nquantiles=50,
+    quartiles=True,
+    point_interval=None,
+    ax=None,
+    show=None,
+    plot_kwargs=None,
+    backend=None,
+    backend_kwargs=None,
+    **kwargs
+):
+
+    """Plot distribution as dot plot or quantile dot plot.
+
+    This function uses the Wilkinson's Algorithm
+    (Leland Wilkinson (1999) Dot Plots, The American Statistician, 53:3, 276-281,
+    DOI: 10.1080/00031305.1999.10474474)to allot dots to bins.
+    The quantile dot plots was inspired from the paper(Matthew Kay, Tara Kola, Jessica R. Hullman,
+    and Sean A. Munson. 2016. When (ish) is My Bus? User-centered Visualizations of Uncertainty in
+    Everyday, Mobile Predictive Systems. DOI:https://doi.org/10.1145/2858036.2858558).
+
+    Parameters
+    ----------
+    values : array-like
+        Values to plot
+    binwidth : float
+        Width of the bin for drawing the dot plot.
+    dotsize : float
+        The size of the dots relative to the bin width. The default, 1, makes dots be
+        just about as wide as the bin width.
+    stackratio : float
+        The distance between the center of the dots in the same stack relative to the bin height.
+        The default, 1, makes dots in the same stack just touch each other.
+    point_interval : bool
+        Plots the point interval. Uses hdi_prob to plot the HDI interval
+    point_estimate : Optional[str]
```

Comment:
```suggestion
    point_estimate : str, optional
```

not all the codebase follows the [numpydoc convention](https://arviz-devs.github.io/arviz/contributing/contributing_prs.html#docstring-formatting-and-type-hints) (the standard to be followed wasn't clear originally), but new code should stick to it now that we have. The `Optional[str]` syntax is for type hints, that go in the function definition, not in the docstring.